mlrun/projects/pipelines.py

```diff
@@ -354,11 +360,17 @@ def enrich_function_object(
         f.enrich_runtime_spec(
             project.spec.default_function_node_selector,
         )
-
+    skip_automount_for_kfp = False
+    if project.spec.workflows:
+        skip_automount_for_kfp = (
+            pipeline_context.workflow.get("engine") == mlrun.EngineType.KFP
```

Comment:
Are you guaranteed that `pipeline_context.workflow` is set when you get here? If all you do is run a function (even though the project has workflows defined), will you have a valid workflow in the context?
I think the `if` should change to `if pipeline_context.workflow:` which is a better indication that we're running in workflow context.